sdk/confidentialledger/azure-security-confidentialledger/src/test/java/com/azure/security/confidentialledger/ConfidentialLedgerClientTestBase.java

```diff
@@ -41,89 +40,83 @@ class ConfidentialLedgerClientTestBase extends TestBase {
 
             confidentialLedgerClientBuilder = new ConfidentialLedgerClientBuilder()
-                    .ledgerEndpoint(Configuration.getGlobalConfiguration().get("LEDGER_URI", "https://emily-java-sdk-tests.confidential-ledger.azure.com"))
-                    .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC));
+                .ledgerEndpoint(Configuration.getGlobalConfiguration().get("LEDGER_URI", "https://emily-java-sdk-tests.confidential-ledger.azure.com"))
+                .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC));
+
             if (getTestMode() == TestMode.PLAYBACK) {
-                confidentialLedgerClientBuilder
-                        .httpClient(interceptorManager.getPlaybackClient())
-                        .credential(request -> Mono.just(new AccessToken("this_is_a_token", OffsetDateTime.MAX)));
+                confidentialLedgerClientBuilder = confidentialLedgerClientBuilder
+                    .httpClient(interceptorManager.getPlaybackClient())
+                    .credential(request -> Mono.just(new AccessToken("this_is_a_token", OffsetDateTime.MAX)));
             } else if (getTestMode() == TestMode.RECORD) {
-                confidentialLedgerClientBuilder
-                        .addPolicy(interceptorManager.getRecordPolicy())
-                        .credential(new DefaultAzureCredentialBuilder().build());
+                confidentialLedgerClientBuilder = confidentialLedgerClientBuilder
+                    .addPolicy(interceptorManager.getRecordPolicy())
+                    .httpClient(httpClient)
+                    .credential(new DefaultAzureCredentialBuilder().build());
             } else if (getTestMode() == TestMode.LIVE) {
-                confidentialLedgerClientBuilder.credential(new AzureCliCredentialBuilder().build());
+                confidentialLedgerClientBuilder = confidentialLedgerClientBuilder
+                    .credential(new DefaultAzureCredentialBuilder().build())
+                    .httpClient(httpClient);
             }
+            confidentialLedgerClient = confidentialLedgerClientBuilder.buildClient();
+
         } catch (Exception ex) {
             System.out.println("Error thrown from ConfidentialLedgerClientTestBase:" + ex);
-        }
-
-        String ledgerId = Configuration.getGlobalConfiguration().get("LEDGER_NAME", "emily-java-sdk-tests");
-        Response<BinaryData> ledgerIdentityWithResponse = confidentialLedgerCertificateClient
-                .getLedgerIdentityWithResponse(ledgerId, null);
-        BinaryData identityResponse = ledgerIdentityWithResponse.getValue();
-        ObjectMapper mapper = new ObjectMapper();
-
-        JsonNode jsonNode = null;
-        try {
-            jsonNode = mapper.readTree(identityResponse.toBytes());
-        } catch (IOException ex) {
-            System.out.println("Caught exception " + ex);
             Assertions.assertTrue(false);
```

Comment:
You can use the `fail()` method to make the test fail here instead of an assertion. Also, would it not be better to throw the exception and not use a try/catch block?

Reply:
Sounds good. I removed the exception handling.